### PR TITLE
[Feature-1906][git] add git project

### LINKED ISSUE #1906: [Feature][git & mvn] New Git and Maven support, and extend the support of UDF（支持git，下载源代码文件并编译，可回归到udf的使用）
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

1. support git 
2. git project -> support java & scala project mvn compile and package
3. git project -> support udf

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zackyoungh**
## Purpose of the pull request

add git project

### CODE REVIEW

**Comment by Zzm0809** [comment]
Comments need to use java doc
and need to use English

**Review by aiwenmo — APPROVED** [review]
LGTM

File: dinky-admin/src/main/java/org/dinky/controller/GitController.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+
+import lombok.AllArgsConstructor;
+
+/**
+ * @author ZackYoung
+ * @since 0.8.0
+ */
+@RestController
+@RequestMapping("/git")
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/controller/GitController.java]: change to /api/git

File: dinky-admin/src/main/java/org/dinky/controller/GitController.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @DeleteMapping("/deleteProject")
+    public Result<Void> deleteProject(Long id) {
+        gitProjectService.removeById(id);
+        return Result.succeed();
+    }
+
+    @PostMapping("/updateState")
+    public Result<Void> updateState(Long id, boolean enable) {
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/controller/GitController.java]: only need id

File: dinky-admin/src/main/java/org/dinky/dto/GitProjectDTO.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    /** */
+    private String username;
+
+    /** */
+    private String password;
+
+    private String privateKey;
+
+    /** */
+    private String pom;
```
Comment (zhu-mingye) [inline dinky-admin/src/main/java/org/dinky/dto/GitProjectDTO.java:52]: has pom ？

File: dinky-admin/src/main/java/org/dinky/dto/GitProjectDTO.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    private String buildArgs;
+
+    /** */
+    private Integer codeType;
+
+    /** */
+    private String description;
+
+    /** */
+    @NotNull private Integer enable;
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/dto/GitProjectDTO.java:64]: change type boolean

File: dinky-admin/src/main/java/org/dinky/model/GitProject.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    @TableField(value = "description")
+    private String description;
+
+    /** */
+    @TableField(value = "build_state")
+    private Integer buildState;
+
+    /** */
+    @TableField(value = "enable")
+    private Integer enable;
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/model/GitProject.java]: boolean

File: dinky-admin/src/main/java/org/dinky/service/GitProjectService.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    void saveOrUpdate(GitProjectDTO gitProjectDTO);
+
+    /**
+     * 更新状态
+     *
+     * @param id id
+     * @param state state
+     */
+    void updateState(Long id, boolean state);
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/service/GitProjectService.java:44]: only need id

File: dinky-admin/src/main/java/org/dinky/service/impl/GitProjectServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        GitProject gitProject = BeanUtil.toBean(gitProjectDTO, GitProject.class);
+        BeanUtil.copyProperties(gitProjectDTO, gitProject);
+
+        gitProject.insertOrUpdate();
+    }
+
+    @Override
+    public void updateState(Long id, boolean state) {
+        GitProject gitProject = baseMapper.selectById(id);
+        gitProject.setEnabled(state);
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/service/impl/GitProjectServiceImpl.java:63]: gitProject.setEnabled(!gitProject.getEnabled());

File: dinky-admin/src/main/resources/mapper/GitProject.xml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+<mapper namespace="org.dinky.mapper.AlertGroupMapper">
+
+    <select id="selectForProTable" resultType="org.dinky.model.GitProject">
+        select
+        a.*
+        from
+        dinky_git_project a
+        <where>
+            1=1
+            <if test='param.name!=null and param.name!=""'>
```
Comment (Zzm0809) [inline dinky-admin/src/main/resources/mapper/GitProject.xml]: need added filters ：eg：Branches， URL，description,codetype, type,buildstatus

File: script/sql/dinky.sql [hunk 9]
```diff
@@ -1046,3 +1046,31 @@ CREATE TABLE dinky_role_select_permissions
...
+                                     `password` varchar(255) DEFAULT NULL,
+                                     `privateKey` varchar(255) DEFAULT NULL COMMENT 'keypath',
+                                     `pom` varchar(255) DEFAULT NULL,
+                                     `build_args` varchar(255) DEFAULT NULL,
+                                     `code_type` tinyint(4) DEFAULT NULL COMMENT 'code type(1-java,2-python)',
+                                     `type` tinyint(4) NOT NULL COMMENT '1-http ,2-ssh',
+                                     `last_build` datetime DEFAULT NULL,
+                                     `description` varchar(255) DEFAULT NULL,
+                                     `build_state` tinyint(4) DEFAULT '-1',
+                                     `enable` tinyint(4) NOT NULL DEFAULT '1' COMMENT '0-disable 1-enable',
```
Comment (Zzm0809) [inline script/sql/dinky.sql]: change to tinyint(1)

File: script/sql/dinky.sql [hunk 10]
```diff
@@ -1046,3 +1046,31 @@ CREATE TABLE dinky_role_select_permissions
...
+                                     `username` varchar(255) DEFAULT NULL,
+                                     `password` varchar(255) DEFAULT NULL,
+                                     `privateKey` varchar(255) DEFAULT NULL COMMENT 'keypath',
+                                     `pom` varchar(255) DEFAULT NULL,
+                                     `build_args` varchar(255) DEFAULT NULL,
+                                     `code_type` tinyint(4) DEFAULT NULL COMMENT 'code type(1-java,2-python)',
+                                     `type` tinyint(4) NOT NULL COMMENT '1-http ,2-ssh',
+                                     `last_build` datetime DEFAULT NULL,
+                                     `description` varchar(255) DEFAULT NULL,
+                                     `build_state` tinyint(4) DEFAULT '-1',
```
Comment (Zzm0809) [inline script/sql/dinky.sql]: has status value? eg: 1:notbuild,2:building,3:buildsuccess, 4:builderror

File: dinky-admin/src/main/java/org/dinky/model/GitProject.java [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    @TableField(value = "build_args")
+    private String buildArgs;
+
+    /** */
+    @TableField(value = "code_type")
+    private Integer codeType;
+
+    /** */
+    @TableField(value = "last_build")
+    private Date lastBuild;
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/model/GitProject.java:78]: this type may be is datetime?

File: dinky-admin/src/main/java/org/dinky/controller/GitController.java [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+    @GetMapping(path = "/build")
+    public Result<Void> build(Long id) {
+
+        GitProject gitProject = gitProjectService.getById(id);
+        if (gitProject.getBuildState().equals(1)) {
+            return Result.failed("此任务正在构建");
+        }
+
+        Dict params = new Dict();
```
Comment (Zzm0809) [inline dinky-admin/src/main/java/org/dinky/controller/GitController.java:104]: 下沉到service层？

**Pull request merged**